### PR TITLE
FED-1254 add Suspense component

### PR DESCRIPTION
**Pull request by kealjones-wk**
Some components are super "lazy", let's add a way to help them out.

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/react_client/component_factory.dart
Comment (kealjones-wk) [inline lib/react_client/component_factory.dart:386]: what the heck? i didn't change these...

File: example/lazy/index.html
Comment (joebingham-wk) [inline example/lazy/index.html:1]: This isn't linked up to the root `index.html`, so you have to manually navigate to it! I think we should add a link since the other stuff is all linked from there

File: lib/react.dart
Comment (joebingham-wk) [inline lib/react.dart:71]: I know it's not going public yet, but when it does, I still think there should be an extended doc comment with consumption guidelines talking about using `Lazy`, or not if it's not public, so folks can easily find how to convert the JS paradigm to Dart! Let me know (whenever you circle back here) if there was a reason that was included when you addressed the first round feedback!

**Review by joebingham-wk — CHANGES_REQUESTED** [review]

File: example/lazy/index.html
Comment (kealjones-wk) [reply example/lazy/index.html:1]: I was actually not showing it quite yet just because `lazy` isn't really like "public" yet... But i guess i could be convinced since it demos suspense... I honestly even considered not including this but i used to for manual testing... and thought it would be a shame to lose it lol

File: lib/react.dart
Comment (kealjones-wk) [reply lib/react.dart:71]: There is a long and lovely doc comment in the over_react version already: https://github.com/Workiva/over_react/pull/820/files#diff-0b321057dde5988d314e11179540acd2207feda3eba1af26f6ac81bd04f603a6R24-R55

eat your heart out

Comment (joebingham-wk) [reply lib/react.dart:71]: > eat your heart out

I would if I felt you addressed my feedback 😜 I won't push it more after this thought, but IMO if we aren't going to put the same level of effort into react-dart as we do OverReact, in terms of comments and API guidelines, then they shouldn't be separate packages. I think the API as is in react-dart is pretty confusing, which is lame for an OSS package

File: example/lazy/index.html
Comment (joebingham-wk) [reply example/lazy/index.html:1]: It's as much a `Suspense` demo as it is a lazy one, and `Suspense` is public 🙂 I think it also just makes our life harder  as well, so I'd still vote for it, but it's probably not worth blocking the PR over!

**Review by joebingham-wk — APPROVED** [review]
Tests, interop, and demo look good!!

+10

File: lib/react.dart
Comment (kealjones-wk) [reply lib/react.dart:71]: bro, same. lol I think react-dart is a waste of time, especially as dart evolves (@greglittlefield-wf probably feels very different) but PERSONALLY it seems like a somewhat unnecessary step between us and the JS, but also there is so much like history there that removing the react-dart dependency would probably break so much shit it would never work lol.

**Review by joebingham-wk — DISMISSED** [review]
+10 refresh, looking awesome!!

File: lib/react.dart
Comment (joebingham-wk) [inline lib/react.dart:75]: Should this be a nested indent?

```suggestion
/// - (dynamic imports, not currently implemented in dart)
```

Comment (kealjones-wk) [reply lib/react.dart:75]: that was intentional as it is a clarification on the point above it.

**Review by joebingham-wk — APPROVED** [review]
+10

**Comment by kealjones-wk** [comment]
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

File: lib/react.dart
Comment (greglittlefield-wf) [reply lib/react.dart:98]: The separation of react-dart and over_react is mostly: 
a) tech debt (these started as separate packages, mainly because we didn't own react-dart)
b) organizationally/architecturally, react-dart works well (for the most part) as a separate layer from over_react, handling most of the JS interop, and over_react mostly being the typed props and factory layer on top of that. Even if we were to combine them into a single repo and make most of the react-dart APIs private, we'd probably maintain most of this separation.

The effort required to maintain the two of them separately hasn't been enough at this point to justify the effort to consolidating them. If we find having them be separate is too painful, I'm happy to discuss!

>  but IMO if we aren't going to put the same level of effort into react-dart as we do OverReact, in terms of comments and API guidelines, then they shouldn't be separate packages. I think the API as is in react-dart is pretty confusing, which is lame for an OSS package

That's a fair point. I think the reality here is that we don't expect people to use react-dart directly most of the time, and instead to use over_react. Perhaps to help address this while these packages are separate, we could just be more forthright about how much effort we put into this project's developer experience, and recommend using over_react instead of using this project directly.

**Pull request merged**
